composeApp/src/commonMain/kotlin/org/noiseplanet/noisecapture/ui/screens/RequestPermissionScreen.kt

```diff
@@ -125,7 +125,7 @@ fun RequestPermissionScreen(
 
                     // If permission state is not yet determined, show a button to trigger
                     // the permission request popup
-                    AnimatedVisibility(permissionState == PermissionState.NOT_DETERMINED) {
+                    AnimatedVisibility(permissionState.notGranted()) {
```

Comment:
If I'm not mistaken, if permission has already been explicitely denied the popup won't show the second time, so showing the request button is no longer relevant. Users should go to the corresponding settings page with the settings button then.